packages/grid/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -363,3 +432,70 @@ export async function buildExcel(
 
   return workbook;
 }
+
+export interface ExcelExportInitEvent {
+  serializedColumns: SerializedColumns;
+  serializedRows: SerializedRow[];
+  valueOptionsSheetName: string;
+  columnGroupPaths: Record<string, string[]>;
+  columnGroupDetails: GridColumnGroupLookup;
+  valueOptionsData: ValueOptionsData;
+  options: Omit<
+    GridExcelExportOptions,
+    | 'workerUrl'
+    | 'exceljsPreProcess'
+    | 'exceljsPostProcess'
+    | 'columnsStyles'
+    | 'valueOptionsSheetName'
+  >;
+}
+
+export function setupExcelExportWebWorker(
+  workerOptions: Pick<GridExcelExportOptions, 'exceljsPostProcess' | 'exceljsPreProcess'> = {},
+) {
+  // eslint-disable-next-line no-restricted-globals
+  addEventListener('message', async (event: MessageEvent<ExcelExportInitEvent>) => {
+    const {
```

Comment:
Could we refactor `buildExcel` to use the serialized data instead of `apiRef` and reuse `buildExcel` in `setupExcelExportWebWorker`?

```tsx
function setupExcelExportWebWorker() {
  addEventListener('message', async (event: MessageEvent<ExcelExportInitEvent>) => {
    const workbook = await buildExcel(event.data);
    postMessage(await workbook.xlsx.writeBuffer());
  }
}
```

Reply:
Sounds good to me 👍